infrastructure/demo/main/ecs.tf

```
@@ -0,0 +1,350 @@
+  # The CPU and Memory may need to be adjusted based on the application usage
+  family             = "admin-app-task"
+  execution_role_arn = aws_iam_role.web_task_role.arn
+  container_definitions = jsonencode([{
+    name   = "admin-app-container",
+    image  = "admin-app:latest",
+    memory = 512,
+    cpu    = 256,
+
+
+
+    portMappings = [
+      {
+        "containerPort" : 3000,
+        "hostPort" : 3000,
+        "protocol" : "tcp"
+      }
+    ]
+
+    logConfiguration = {
+      logDriver = "awslogs"
+      options = {
+        awslogs-group         = aws_cloudwatch_log_group.admin_app.name
+        awslogs-stream-prefix = "ecs"
+        awslogs-region        = var.aws_region
+      }
+    }
+  }])
+
+}
+
+# Backend Service with EC2 Launch Type
+resource "aws_ecs_service" "backend_service" {
+  # This is a resource, which means it will create a resource in AWS
+  # This resource will create an ECS service with EC2 launch type for the Backend container
+  # The ECS service will be used to route traffic to the Backend container
+  # The ECS service will be attached to the ECS cluster
+  name                               = "backend-service"
+  cluster                            = aws_ecs_cluster.web_cluster.id
+  task_definition                    = aws_ecs_task_definition.backend_task.arn
+  deployment_minimum_healthy_percent = 0
+  deployment_maximum_percent         = 200
+  launch_type                        = "EC2"
+  scheduling_strategy                = "REPLICA"
+  desired_count                      = 1
+
+  service_registries {
+    registry_arn   = aws_service_discovery_service.backend.arn
+    container_name = "backend-container"
+    container_port = 8000
+  }
+
+  lifecycle {
+    ignore_changes = [task_definition, desired_count]
+  }
+}
+
+resource "aws_ecs_task_definition" "backend_task" {
+  # The rest of the container definitions will be added when the application is deployed. It will be added to the task definition from docker-compose.yml using the ecs-cli compose create command
+  family             = "backend-task"
+  execution_role_arn = aws_iam_role.web_task_role.arn
+  container_definitions = jsonencode([{
+    name   = "backend-container",
+    image  = "backend:latest",
+    memory = 512,
+    cpu    = 256,
+
+    portMappings = [
+      {
+        "containerPort" : 8000,
+        "hostPort" : 8000,
+        "protocol" : "tcp"
+      }
+    ]
+
+    logConfiguration = {
+      logDriver = "awslogs"
+      options = {
+        awslogs-group         = aws_cloudwatch_log_group.backend.name
+        awslogs-stream-prefix = "ecs"
+        awslogs-region        = var.aws_region
+      }
+    }
+  }])
+
+
+
+
+
+}
+
+resource "aws_cloudwatch_log_group" "admin_app" {
+  name = "/ecs/admin-app-task-${var.environment}"
+
+  tags = merge({ Name = "admin-app-task-${var.environment}", Module = "Web" }, var.tags)
```

We need to add project name to the names also. Otherwise can't create another project resource in the same AWS account!
```suggestion
  name = "/ecs/admin-app-task-${var.project_name}-${var.environment}"

  tags = merge({ Name = "admin-app-task-${var.project_name}-${var.environment}", Module = "Web" }, var.tags)
```